plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -321,6 +323,11 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 	if (did == 0x0)
 		g_warning ("failed to get Device ID: %s", error_did->message);
 
+	/* requires kernel 5.5 or later, non-fatal if not available */
+	gen = fu_plugin_thunderbolt_udev_get_id (device, "generation", &error_gen);
+	if (gen == 0)
+		g_debug ("Unable to read generation: %s", error_gen->message);
```

Comment:
Since we are using `fu_plugin_thunderbolt_udev_get_id` in the very unlikely case of overflow, the message will contain `vendor id overflows`. I guess we could (and should) rename the function later to make it more general purpose.

Reply:
Sure, good point.  I just pushed a fixup that does that.